web/src/HelpSearchBar.tsx

```diff
@@ -0,0 +1,114 @@
+import { InputAdornment } from "@material-ui/core"
+import { InputProps as StandardInputProps } from "@material-ui/core/Input/Input"
+import React, { ChangeEvent, KeyboardEvent, useState } from "react"
+import styled from "styled-components"
+import { ReactComponent as CloseSvg } from "./assets/svg/close.svg"
+import { ReactComponent as SearchSvg } from "./assets/svg/search.svg"
+import {
+  InstrumentedButton,
+  InstrumentedTextField,
+} from "./instrumentedComponents"
+import {
+  Color,
+  Font,
+  FontSize,
+  mixinResetButtonStyle,
+  SizeUnit,
+} from "./style-helpers"
+
+export function searchDocs(query: string) {
+  window.open(
+    `https://docs.tilt.dev/search?q=${encodeURI(query)}&utm_source=tiltui`
+  )
+}
+
+export const HelpSearchBarTextField = styled(InstrumentedTextField)`
+  & .MuiOutlinedInput-root {
+    border-radius: ${SizeUnit(0.5)};
+    background-color: ${Color.white};
+
+    & fieldset {
+      border-color: 1px solid ${Color.grayLighter};
+    }
+    &:hover fieldset {
+      border: 1px solid ${Color.grayLighter};
+    }
+    &.Mui-focused fieldset {
+      border: 1px solid ${Color.grayLighter};
+    }
+    & .MuiOutlinedInput-input {
+      padding: ${SizeUnit(0.2)};
+    }
+  }
+
+  margin-top: ${SizeUnit(0.4)};
+  margin-bottom: ${SizeUnit(0.4)};
+
+  & .MuiInputBase-input {
+    font-family: ${Font.monospace};
+    color: ${Color.grayLighter};
+    font-size: ${FontSize.small};
+  }
+`
+
+export const ClearHelpSearchBarButton = styled(InstrumentedButton)`
+  ${mixinResetButtonStyle};
+  display: flex;
+  align-items: center;
+`
+export let searchValue: string
+let setSearchValue: React.Dispatch<React.SetStateAction<string>>
```

Comment:
Yes, that's precisely why. Would definitely appreciate the alternative approach because I don't feel good about this 😅

Reply:
for sure! there are two ways you could do this:

```ts
// Grab the `input` DOM node and check its `value` property
const searchFieldAfterInput = root.find("input").at(0)
expect((searchFieldAfterInput.getDOMNode() as HTMLInputElement).value).toEqual(searchTerm)

// Grab the text field component and check its `value` prop
const searchComponentAfterInput = root.find(HelpSearchBarTextField).at(0)
expect(searchComponentAfterInput.prop("value")).toEqual(searchTerm)
```

for both approaches, you need to `find` the elements in the wrapper that you want to access after the component has updated. i've forgotten about this in the past. 😅 

```ts
// Fetch the element
const searchField = root.find("input")
// Make the change
searchField.simulate("change", { target: { value: searchTerm } })
// To check something in the component affected by the change, fetch it again
const searchFieldAfterInput = root.find("input")
```

(i usually have the [enzyme docs](https://enzymejs.github.io/enzyme/docs/api/mount.html) bookmarked because i forget all of the helpers it has to access component / node properties.)